MoreLinq.Test/SegmentTest.cs

```diff
@@ -146,5 +148,29 @@ public void VerifyCanSegmentByPrevious()
             Assert.AreEqual(sequence.Distinct().Count(), result.Count());
             Assert.IsTrue(result.All(s => s.Count() == repCount));
         }
+
+        static IEnumerable<T> Seq<T>(params T[] values) => values;
+
+        public static readonly IEnumerable<ITestCaseData> TestData =
+            from e in new[]
+            {
+                // input sequence is empty
+                new { Source = Enumerable.Empty<int>(), Expected = Enumerable.Empty<IEnumerable<int>>() },
+                // input sequence contains only new segment start
+                new { Source = Seq(0, 3, 6),            Expected = Seq(Seq(0), Seq(3), Seq(6)) },
+                // input sequence do not contains new segment start
+                new { Source = Seq(1, 2, 4, 5),         Expected = Seq(Seq(1, 2, 4, 5)) },
+                // input sequence start with a segment start
+                new { Source = Seq(0, 1, 2, 3, 4, 5),   Expected = Seq(Seq(0, 1, 2), Seq(3, 4, 5)) },
+                // input sequence do not start with a segment start
+                new { Source = Seq(1, 2, 3, 4, 5),      Expected = Seq(Seq(1, 2), Seq(3, 4, 5)) }
+            }
+            select new TestCaseData(e.Source).Returns(e.Expected);
+
+        [Test, TestCaseSource(nameof(TestData))]
+        public IEnumerable<IEnumerable<int>> TestSegmentOnKnownInput(IEnumerable<int> source)
```

Comment:
Simplify name to just:

```suggestion
        public IEnumerable<IEnumerable<int>> TestSegment(IEnumerable<int> source)
```